z3c/dependencychecker/db.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ImportsDatabase(object):
+
+    def __init__(self):
+        self._requirements = []
+        self._extras_requirements = {}
+
+    def set_requirements(self, requirements_list):
```

Comment:
Shouldn't it be `add_requirements()`? That is, you're adding items to `_self._requirements`. Sounds like it can happen more than once?
OTOH, that isn't really logical.
But.... then it would make much more sense to say something like `self._requirements = set(requirements_list)`.

Reply:
Normally, I dislike names with "list" or "dict" in them. Sometimes it has its uses, but here it seems a bit odd.

Reply:
Actually requirements_list is quite misleading because, actually is an iterator, so I changed that to a plain requirements.

I thought that I did the extra effort to mimic the current z3c.dependencychecker implementation, but a quick search seems that it does not have it, so I will remove it.

To not block the flow of pull requests, if that's ok with you (there are at maximum 2 more to come), I will do a full switch to properties (or cached properties actually) after all code is merge, so that we keep everything consistent style wise (or at least as much consistent as I achieved to be).

I renamed the method to ``add_requirements`` but as mentioned above, we should move that to a property later on.